### PR TITLE
fix: carousel's pauseOnDotsHover not work

### PR DESCRIPTION
**Pull request by snowtreetree**
### This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Use Carousel, set autoplay, autoplaySpeed, pauseOnDotsHover.
> 2. Hover on dots, it doesn't work.

### API Realization (Optional if not new feature)

> 1.  show file changes.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. fix Carousel's prop of pauseOnDotsHover not work.
> 2. 修复Carousel组件pauseOnDotsHover失效问题。

### Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by ajuner** [comment]
确认一下吧，试了一下pauseOnDotsHover是生效的

**Comment by snowtreetree** [comment]
https://codesandbox.io/s/adoring-butterfly-1jxji?file=/src/App.vue
@ajuner

**Comment by snowtreetree** [comment]
https://user-images.githubusercontent.com/30932683/104266611-ec543280-54ca-11eb-8433-caf24b2457ba.mov

**Comment by ajuner** [comment]
你这个demo是1.x版本的吧。。

**Comment by snowtreetree** [comment]
https://codesandbox.io/s/nostalgic-water-juux5?file=/src/main.js
这个是2.0.0-rc.6版本的，你这边确认的时候是用的什么版本，😂。

**Comment by ajuner** [comment]
https://codesandbox.io/s/pensive-nash-56xzn?file=/src/App.vue
神奇的问题，我这个是能停住的

**Comment by snowtreetree** [comment]
image
@ajuner 好像单独用是可以的，再在外面多加一层就不行了。

File: components/vc-slick/src/inner-slider.js [hunk 1]
```diff
@@ -635,7 +635,7 @@ export default {
       dotProps = {
         ...dotProps,
         clickHandler: this.changeSlide,
-        onMouseenter: pauseOnDotsHover ? this.onDotsLeave : noop,
+        onMouseenter: pauseOnDotsHover ? this.onDotsOver : noop,
```
Comment (tangjinzhou) [inline components/vc-slick/src/inner-slider.js:638]: I think we can delete onMouseenter event.

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

**Comment by tangjinzhou** [comment]
删掉那一行应该没什么问题

File: components/vc-slick/src/inner-slider.js (hunk 1 above)
Comment (snowtreetree) [reply components/vc-slick/src/inner-slider.js:638]: 确实，这块onMouseover和onMouseenter处理情况有重复的地方。

Comment (snowtreetree) [reply components/vc-slick/src/inner-slider.js:638]: pauseOnHover 相关的 onMouseenter应该也可以删除，😂。

**Pull request merged**